src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -3741,10 +3741,10 @@ private bool ParseXmlDeclaration(bool isTextDecl)
                 int nameEndPos = ParseName();
 
                 NodeData? attr = null;
-                switch (_ps.chars[_ps.charPos])
+                switch (_ps.chars.AsSpan(_ps.charPos..nameEndPos))
```

Comment:
Using `Range` (which this syntax does) is slightly more expensive than just using the two-arg version of AsSpan.  I have a small preference for this instead being:
```suggestion
                switch (_ps.chars.AsSpan(_ps.charPos, nameEndPos - _ps.charPos))
```